core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -307,42 +307,40 @@ object ConfigCommand extends Config {
+      ++ configsToBeDeleted.map { k => new AlterConfigOp(new ConfigEntry(k, ""), AlterConfigOp.OpType.DELETE) }
+    ).asJavaCollection
 
     entityType match {
       case ConfigType.Topic =>
-        val oldConfig = getConfig(adminClient, entityType, entityName, includeSynonyms = false, describeAll = false)
-          .map { entry => (entry.name, entry) }.toMap
-
         // fail the command if any of the configs to be deleted does not exist
+        val oldConfig = getConfig(adminClient, ConfigType.Topic, entityName, includeSynonyms = false, describeAll = false)
+          .map { entry => (entry.name, entry) }.toMap
         val invalidConfigs = configsToBeDeleted.filterNot(oldConfig.contains)
+
         if (invalidConfigs.nonEmpty)
           throw new InvalidConfigurationException(s"Invalid config(s): ${invalidConfigs.mkString(",")}")
 
         val configResource = new ConfigResource(ConfigResource.Type.TOPIC, entityName)
-        val alterOptions = new AlterConfigsOptions().timeoutMs(30000).validateOnly(false)
-        val alterEntries = (configsToBeAdded.values.map(new AlterConfigOp(_, AlterConfigOp.OpType.SET))
-          ++ configsToBeDeleted.map { k => new AlterConfigOp(new ConfigEntry(k, ""), AlterConfigOp.OpType.DELETE) }
-        ).asJavaCollection
         adminClient.incrementalAlterConfigs(Map(configResource -> alterEntries).asJava, alterOptions).all().get(60, TimeUnit.SECONDS)
 
       case ConfigType.Broker =>
-        val oldConfig = getConfig(adminClient, entityType, entityName, includeSynonyms = false, describeAll = false)
-          .map { entry => (entry.name, entry) }.toMap
-
         // fail the command if any of the configs to be deleted does not exist
+        val oldConfig = getConfig(adminClient, ConfigType.Broker, entityName, includeSynonyms = false, describeAll = false)
+          .map { entry => (entry.name, entry) }.toMap
         val invalidConfigs = configsToBeDeleted.filterNot(oldConfig.contains)
+
         if (invalidConfigs.nonEmpty)
           throw new InvalidConfigurationException(s"Invalid config(s): ${invalidConfigs.mkString(",")}")
 
         val newEntries = oldConfig ++ configsToBeAdded -- configsToBeDeleted
         val sensitiveEntries = newEntries.filter(_._2.value == null)
         if (sensitiveEntries.nonEmpty)
           throw new InvalidConfigurationException(s"All sensitive broker config entries must be specified for --alter, missing entries: ${sensitiveEntries.keySet}")
-        val newConfig = new JConfig(newEntries.asJava.values)
 
         val configResource = new ConfigResource(ConfigResource.Type.BROKER, entityName)
-        val alterOptions = new AlterConfigsOptions().timeoutMs(30000).validateOnly(false)
-        adminClient.alterConfigs(Map(configResource -> newConfig).asJava, alterOptions).all().get(60, TimeUnit.SECONDS)
+        adminClient.incrementalAlterConfigs(Map(configResource -> alterEntries).asJava, alterOptions).all().get(60, TimeUnit.SECONDS)
```

Comment:
Could we use [Admin.incrementalAlterConfigs(Map)](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/admin/Admin.java#L407)? the default AlterConfigsOptions should be fine to this tool.